### PR TITLE
Reviewer: Don't blink action bar icons

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
User Report: It's distracting for the action item buttons to blink when answering cards.

This was added in 58ec2523777845955f3f7bfbb375c44c4339e818 (#5897)

## Approach
Partial revert of the above commit

## How Has This Been Tested?
On my phone, icons no longer blink

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by david-allison-1** [comment]
@Arthur-Milchior pinging as this will impact your workflow.

I feel it's better to optimise for the majority of users, but this is fully open for discussion to find a better solution/compromise.

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/anki/reviewer/ActionButtonStatus.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5553069&bid=18207166)

**Comment by david-allison-1** [comment]
Converted to draft. 

Opening discussion.

I can't see a good way to solve it, but I don't like the visual flicker.

TL;DR: Predicting future events is hard

It's not going to be easily possible, as we want to measure the time between an event that has occurred, and a future event. If we wait 100ms before disabling the button, the result could be returned at the 101th ms, causing the flicker.

Adding a reasonable wait time before disabling would work for the majority of cases, but aside from the flicker afterwards, we'd still have the issue that the user clicks the button before the 100ms mark, and and doesn't get visual feedback that the button is disabled.

**Comment by mikehardy** [comment]
I would give it a 250ms budget at least https://www.pubnub.com/blog/how-fast-is-realtime-human-perception-and-technology/

It's not a twitch game we're playing and the user won't be able to move their finger to the app bar and attempt anything within a quarter second minimum (I'd argue it could be higher, like 500ms). Set a delayed action to alpha disable them for 500ms in the future and if the async task returns cancel it?

Not sure how complicated that is but I think we have a lot of room here that results in no flicker for almost all cases, but a still "accurate" UI as far as human perception goes

**Comment by Arthur-Milchior** [comment]
> TL;DR: Predicting future events is hard

Let me disagree with you. There are actions which are known to be slow and other which are known to be quick.
In my personal experience, what is slow is to rebuild the queue. When you already have a queue of card to review, showing it is usually quick. Of course, there are exceptions, in particular if you card have a lot of mathjax, if you need to load big media. However, media loading and mathjax does not change the behaviour of the buttons so we can ignore those cases.

The queue is rebuilt in two occasions: when you review a card and the queue is empty, (Some pending PR actually pre-fill the queue, so that no more time is lost here) and when you cancel/suspend/bury.
I suggest that we ignore the case where the queue is empty, at least until the pending PRs are discussed in details. If you reject my PRs then we may want to consider how to deal with empty queues.
It remains the case where we cancel, suspend, bury... (I believe this is collectively considered by the word "dismiss")
We can simply state that the buttons gets get set to ALPHA_ICON_DISABLED_LIGHT only when we are loading the first card, or when we "dismissed" a card. 

If some user see that there are case we forgot, where we have to wait a lot, and the buttons are not disabled, then we cans implement the rule of waiting .5 seconds. However, I must admit that i would prefer not having this rule first; mainly because it would add some asynchrone part to the code, and so it would make it even more complx

**Comment by Arthur-Milchior** [comment]
if you agree with this compromise, I can get a look at it this week-end. I hope it's okay, and am sorry for the time lost with it. Mike, I appreciate that you accept that we keep this feature; even if it's not perfect yet. Because, honestly, as a user, I hated when I wanted to cancel twice and had no clue about whether the buttons i clicked were registered or not

**Comment by mikehardy** [comment]
Hmm - maybe just if someone taps the buttons while they are blocked we toast "still processing previous request...". No alpha change so no blink. Most users shouldn't ever encounter that they double-tap because things are slow (I think?) but the toast would be concrete feedback and satisfy me. Quick and easy?

**Comment by Arthur-Milchior** [comment]
In my experience, clicking twice on "cancel" because you want to cancel two actions is something that I naturally do. I would be surprised to be the only one doing it; so I would expect other people to have the same problem.
Furthermore, it occurs too often that I bury/suspend a note by accident and so immediately try to cancel in order to get the card back.


Going with your idea should really easy to do. We only need to look at the method `getControlBlocked`.  Currently it is used only for two reason: deciding whether to do an action, and what alpha to use. This PR removes the second usage. So essentially, `getControlBlocked` is simply a function which state whether an action can be done or not. If the action can't be done, the toast can appear. 

I should admit that I have a preference for changing the alpha when I expect action taking time to occur and not changing it during review. But that is a personal preference, and you're the mainteneur.

**Comment by david-allison-1** [comment]
I prefer #6227, no point in keeping this around if there's a better solution.
